File: adobe-edge/src/api/details/AdobeImplementationDetails.ts
/**
 * Details about the SDK, library, or service used in an application or web page implementation of a service.
 *
 * https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/implementationdetails.schema.md
 */
export interface AdobeImplementationDetails {
  // The environment of the implementation
  environment?: AdobeEnvironment;

  // SDK or endpoint identifier. All SDKs or endpoints are identified through a URI, including extensions.
  name?: string;

  // The version identifier of the API, e.g h.18.
  version?: string;
}

/**
 * The environment of the implementation.
 */
export enum AdobeEnvironment {
  BROWSER = 'browser',
  APP = 'app',
  SERVER = 'server',
  IOT = 'iot',
}

File: adobe-edge/android/src/main/java/com/theoplayer/reactnative/adobe/edge/api/AdobeImplementationDetails.kt
package com.theoplayer.reactnative.adobe.edge.api

/**
 * Details about the SDK, library, or service used in an application or web page implementation of a service.
 *
 * @see <a href="https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/implementationdetails.schema.md">Adobe XDM ImplementationDetails Schema</a>
 */
data class AdobeImplementationDetails(
  // The environment of the implementation
  val environment: AdobeEnvironment? = null,
  // SDK or endpoint identifier. All SDKs or endpoints are identified through a URI, including extensions.
  val name: String? = null,
  // The version identifier of the API, e.g h.18.
  val version: String? = null
)

/**
 * The environment of the implementation.
 */
enum class AdobeEnvironment {
  BROWSER,
  APP,
  SERVER,
  IOT
}
